Ray_Serve/Ray-Serve.ipynb

```diff
@@ -0,0 +1,1897 @@
+    "We can test it as an HTTP endpoint or by invoking directly from Python."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "78ffe466-169f-4d73-9e3c-019d8c62fc2a",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "sample_json = '{ \"input\" : \"hello\" }'\n",
+    "requests.post(\"http://localhost:8000/\", json = sample_json).json()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "89d6a44f-a3a8-426f-8367-0393fb4952a6",
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-block alert-success\">\n",
+    "    \n",
+    "__Lab activity: implement a web service with Ray Serve__\n",
+    "    \n",
+    "The following function will calculate the approximate loan payment for a car.\n",
+    "    \n",
+    "```python\n",
+    " def monthly_payment(total_price, rate, years_of_loan):\n",
+    "    n = 365.25 # compounding periods\n",
+    "    total_paid = total_price * (((1 + ((rate/100.0)/n)) ** (n*years_of_loan)))\n",
+    "    per_month = total_paid / (12 * years_of_loan)\n",
+    "    return per_month\n",
+    "```\n",
+    "   \n",
+    "<br/>\n",
+    "Deploy this calculator as a web service with Ray Serve!\n",
+    "    \n",
+    "</div>\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b40134f3-a5ba-4554-95f9-74a7f7731e66",
+   "metadata": {},
+   "source": [
+    "## Key APIs and concepts\n",
+    "\n",
+    "`Deployment` represents a service and is created with the `@serve.deployment` decorator\n",
+    "* As end users, we don't instantiate `Deployment`s directly\n",
+    "* Ray will create them as actors, per our scaling requirements\n",
+    "\n",
+    "A __bound deployment__ is created with the `.bind` class method on the deployment class\n",
+    "* e.g., `Chat.bind(msg=\"Yes...\")` above creates a bound deployment\n",
+    "* `.bind` allows us to provide constructor params for the deployment class (the `msg` param above)\n",
+    "* bound deployments *can* be passed to other deployments via `.bind` -- this is one way to compose services\n",
+    "* We can pass a bound deployment to `serve.run(...)`\n",
+    "    * to start a service\n",
+    "    * to obtain a `ServeHandle`\n",
+    "\n",
+    "A `ServeHandle` can be used to invoke services through the Python API\n",
+    "* At runtime, services can call other services via serve handles\n",
+    "    * Bound deployments provided to deployment constructors via `.bind` become serve handles at runtime"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e50986b5-b5d5-48e1-b7ce-d5fa27a2121e",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "print(type(handle))\n",
+    "print(handle)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "03b390c1-38fb-40fc-b81f-8b5fcd2c378c",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "response = handle.get_response.remote('hello')\n",
```

Comment:
It would be better demonstration to use http request/response here instead of handle call (since this is meant to be the ingress deployment)